File: Data_Structures_And_Algorithms-UC_SanDiego/1-Algorithmic_Toolbox/Week3/Programming_Assignments/1_money_change/change.py
# Uses python3
import sys

def get_change(m):
    n = 0
    while m:
        if m//10!=0:
            m -= 10
            n += 1
        elif m//5!=0:
            m -= 5
            n += 1
        else:
            m -= 1
            n += 1
    return n

if __name__ == '__main__':
    m = int(input())
    print(get_change(m))

File: Data_Structures_And_Algorithms-UC_SanDiego/1-Algorithmic_Toolbox/Week3/Programming_Assignments/1_money_change/change.cpp
#include <iostream>

int get_change(int m) {
  int n = 0;
  while (m>0){
    if ((int)m/10!=0) {
      n++;
      m -= 10;
    } else if ((int)m/5!=0){
      n++;
      m -= 5;
    } else {
      n++;
      m -= 1;
    }
  }
  return n;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
}
